homeassistant/helpers/entity_component.py

```diff
@@ -85,17 +86,22 @@ def _setup_platform(self, platform_type, platform_config,
             return
 
         # Config > Platform > Component
-        scan_interval = platform_config.get(
-            CONF_SCAN_INTERVAL,
-            getattr(platform, 'SCAN_INTERVAL', self.scan_interval))
+        scan_interval = (platform_config.get(CONF_SCAN_INTERVAL) or
+                         getattr(platform, 'SCAN_INTERVAL', None) or
+                         self.scan_interval)
         entity_namespace = platform_config.get(CONF_ENTITY_NAMESPACE)
 
+        key = (platform_type, scan_interval, entity_namespace)
```

Comment:
In our config we have two entries: light platform X with interval A and platform X with interval B. In that case interval B will be disregarded because it will load the platform entry that came with platform X with interval A, which is wrong.